ddtrace/internal/datastreams/kafka.py

```diff
@@ -13,6 +23,29 @@ def dsm_kafka_message_produce(instance, args, kwargs):
     headers[PROPAGATION_KEY] = encoded_pathway
     kwargs["headers"] = headers
 
+    on_delivery_kwarg = "on_delivery"
+    on_delivery_arg = 5
+    on_delivery = None
+    try:
+        on_delivery = get_argument_value(args, kwargs, on_delivery_arg, on_delivery_kwarg)
+    except ArgumentError:
+        on_delivery_kwarg = "callback"
+        on_delivery_arg = 4
+        on_delivery = get_argument_value(args, kwargs, on_delivery_arg, on_delivery_kwarg, optional=True)
```

Comment:
nit: Instead of catching an argument error can we parse the arguments based on the version of `confluent_kafka` library. 

ex:
```suggestion
   if confluent_kafka.VERSION >= (2,0): #random example not sure if this is the right version
        on_delivery = get_argument_value(args, kwargs, 5, "on_delivery", True)
    else:
        on_delivery = get_argument_value(args, kwargs, 4, "callback", optional=True)
```

Reply:
I'm not entirely sure on the reasoning here; I can dig in some